src/services/renown/utils.ts

```diff
@@ -1,16 +1,23 @@
 import { User } from './types';
 
-export function parsePkhDid(
-    did: string,
-): Pick<User, 'networkId' | 'chainId' | 'address'> {
+export function parsePkhDid(did: string): Pick<User, 'chainId' | 'address'> {
     const parts = did.split(':');
     if (!did.startsWith('did:pkh:') || parts.length !== 5) {
         throw new Error('Invalid pkh did');
     }
+    const [, , , chainIdStr, address] = parts;
+
+    if (!address.startsWith('0x')) {
+        throw new Error(`Invalid address: ${address}`);
+    }
+
+    const chainId = Number(chainIdStr);
+    if (isNaN(chainId)) {
+        throw new Error(`Invalid chain id: ${chainIdStr}`);
+    }
 
     return {
-        networkId: did.split(':')[2],
-        chainId: parseInt(did.split(':')[3]),
-        address: did.split(':')[4],
+        chainId,
+        address: address as `0x${string}`,
```

Comment:
just for future reference, viem and wagmi export a convenience type `Address` that is `0x${string}`